src/test/java/com/devonfw/sample/archunit/ArchitectureTest.java

```diff
@@ -16,15 +16,14 @@ public class ArchitectureTest {
   @ArchTest
   private static final ArchRule shouldOnlyAccessValidLayers = //
       layeredArchitecture().consideringAllDependencies() //
-          .layer("common").definedBy("..common..") //
-          .layer("logic").definedBy("..logic..") //
-          .layer("dataaccess").definedBy("..dataaccess..") //
-          .layer("service").definedBy("..service..") //
-          .layer("batch").definedBy("..batch..") //
-          .layer("ui").definedBy("..ui..") //
-          // TODO
-          .because("Dependency of technical layers violates architecture rules.");
+          .layer("common").definedBy("com.devonfw.sample.archunit.common..") //
+          .layer("logic").definedBy("com.devonfw.sample.archunit.logic..") //
+          .layer("dataaccess").definedBy("com.devonfw.sample.archunit.dataaccess..") //
+          .layer("service").definedBy("com.devonfw.sample.archunit.service..") //
+          .layer("client").definedBy("com.devonfw.sample.archunit.client..")
 
+          .withOptionalLayers(true)
```

Comment:
aha, nice that you found this one to prevent the error if layer is empty :) Great 👍